src/include/duckdb/parser/parsed_data/create_index_info.hpp

```diff
@@ -25,6 +25,8 @@ struct CreateIndexInfo : public CreateInfo {
 	IndexType index_type;
 	//! Name of the Index
 	string index_name;
+	//! Name of the Index type
+	string index_type_name;
```

Comment:
Why do we need this field? Isn't this information contained in `IndexType` already?

Reply:
My idea was that if you have multiple extensions pushing indexes they can differentiate by checking the index name. We can't really extend the IndexType enum from an extension (maybe an extension could just add a random uint8_t and cast it, but that feels dirty, and you would still have to do that at the parsing stage to figure out what kind of index is created)

Reply:
> We can always consider not persisting extension indexes and rebuilding them during startup... And just implementing a very efficient physical index creation operator for the extension, idk... Part of the reason we decided to serialize/lazily load the ART is that it is our primary index structure.

Yeah that's my plan for now too, we can think about persistence later.

Reply:
If we want to add support for persisting custom indexes in our storage format it might be worth at least thinking about before the next release

Reply:
I agree. I will consider the possibility when reworking the serialization. It might also be worth scheduling a short meeting for this. But I'd like to think about it/work a bit on the persistence of the ART first.

We might want to generalize the fixed-size allocators more. So that different indexes can use them for persisting their memory... I'll keep this in mind!